packages/extension-ui/src/components/Toast/ToastProvider.tsx

```diff
@@ -11,32 +12,58 @@ interface ToastProviderProps {
   children?: React.ReactNode;
 }
 
-export const TOAST_TIMEOUT = 1500;
+export const TOAST_TIMEOUT = 2500;
 
 const ToastProvider = ({ children }: ToastProviderProps): React.ReactElement<ToastProviderProps> => {
   const [content, setContent] = useState('');
   const [visible, setVisible] = useState(false);
   const [type, setType] = useState<SnackbarTypes>('info');
+  const method = useRef<NodeJS.Timeout>();
 
-  const show = useCallback((message: string, type: SnackbarTypes = 'info'  ): () => void => {
-    const timerId = setTimeout(() => setVisible(false), TOAST_TIMEOUT);
+  const cancelCallback = useCallback(() => {
+    setVisible(false);
+    clearTimeout(method.current);
+    method.current = undefined;
+  }, [method]);
 
-    setContent(message);
-    setVisible(true);
+  const show = useCallback(
+    (message: string, type: SnackbarTypes = 'info', callback?: () => void): (() => void) => {
+      if (visible) {
+        return noop;
+      }
 
-    if(type){
-      setType(type);
-    }
+      if (callback) {
+        method.current = setTimeout(() => callback?.(), TOAST_TIMEOUT);
```

Comment:
I don't think you need `?` as it check for undefined and you're doing `if callback` so you know it's not undefined ?